docs/proposals/metadata.md

```diff
@@ -0,0 +1,118 @@
+resource templates, like PodTemplates, ReplicationControllers, BuildConfigs
+etc.  Automatic generation of a PodTemplate is also required for
+Deployments.
+
+## Constraints and Assumptions
+
+For the time being, there is no good way to define metadata for Docker
+images other than specifying extra environment variables. There is upstream
+work currently being done to fix this situation and we should switch to that in
+the implementation in the future.
+
+This document only defines the metadata needed by the current set of use cases.
+Additional metadata and/or use cases may be added in the future.
+
+## Use cases
+
+1. As an author of Docker image that is going to be consumed by the OpenShift
+   platform, I want to express what categories/tags my Docker image will belongs
+   to, so the platform then can then consume the tags to improve the generation
+   workflow based on them.
+
+2. As an user of OpenShift I want to get reliable suggestions about services
+   that the Docker image I'm going to use might require to operate properly. As
+   an author of the Docker image, I need to have a way to record what services my
+   Docker image want to consume.
+
+2. As an author of the Docker image, I need to have a way to indicate whether
+   the container started from my Docker image does not support scaling.
+   The UI should then communicate this information with the end consumers.
+
+## Metadata variables (required for 3.0)
+
+### IMAGE_TAGS
+
+This metadata variable contains a list of tags. The tags are the way to
+categorize Docker images into broad areas of functionality. Tags helps UI and
+generation tools to suggest relevant Docker images during the application
+creation process.
+
+Example format:
+
+```
+IMAGE_TAGS=database,mysql
+```
+
+### IMAGE_WANTS
+
+Specifies a list of tags that the generation tools and UI might use for
+suggesting to provide if you don't have the Docker images with given tags
+already. For example, if the Docker image WANTS 'mysql' and 'redis' and you
+don't have the Docker image with 'redis' tag, then UI might suggest you to add
+this image into your application.
+
+The format of the values is the same as for TAGS.
+
+### IMAGE_DESCRIPTION
+
+This variable is used to give the image consumers more detailed description
+about the service or functionality given image provides.
+The UI can then use this description together with the image name to give users
+more informations about the image.
+
+Example:
+
+```
+IMAGE_DESCRIPTION="MySQL 5.5 database"
+```
+
+### IMAGE_EXPOSE_SERVICES
+
+This variable contains a list of ports that match with the EXPOSE instructions
+in Dockerfile and provide more descriptive information about what actual service
+the given port provides.
+
+The format is `PORT[/PROTO]:SERVICE_NAME` where the `[PROTO]` part is optional
+and it defaults to 'tcp' if it is not specified.
+
+Example:
+
+```
+IMAGE_EXPOSE_SERVICES="3128/tcp:mysql,8080:http"
```

Comment:
I don't think this is right.  I would prefer IMAGE_EXPOSE_<port>_PROTOCOLS="protocol1,protocol2,...".